L1Trigger/TrackFindingTracklet/plugins/ProducerKFin.cc

```diff
@@ -113,173 +121,37 @@ namespace trklet {
-            // cut on phi and z residuals
-            if (abs(phi) > rangePhi / 2. || abs(z) > rangeZ / 2.)
-              continue;
-            // layers consitent with rough r-z track parameters are counted from 0 onwards
-            int layer = distance(le.begin(), find(le.begin(), le.end(), setup_->layerId(ttStubRef)));
-            // put stubs from layer 7 to layer 6 since layer 7 almost never has stubs
-            if (layer >= setup_->numLayers())
-              layer = setup_->numLayers() - 1;
-            // cut on max 4 stubs per layer
-            if (layerCounts[layer] >= setup_->zhtMaxStubsPerLayer())
-              continue;
-            layerCounts[layer]++;
-            const StubKFin stubKFin(ttStubRef, dataFormats_, rphi, phi, z, dPhi, dZ, layer);
-            streamsStubs[layer].emplace_back(stubKFin.frame());
-          }
-          const int size = *max_element(layerCounts.begin(), layerCounts.end());
-          for (int layer = 0; layer < setup_->numLayers(); layer++) {
-            deque<FrameStub>& stubs = streamsStubs[layer];
-            const int nGaps = size - layerCounts[layer];
-            stubs.insert(stubs.end(), nGaps, FrameStub());
-          }
-          const TTBV& maybePattern = layerEncoding_->maybePattern(binEta, binZT, binCot);
-          const TrackKFin track(ttTrackRef, dataFormats_, maybePattern, phiT, inv2R, zT, cot, sectorPhi, binEta);
-          streamTracks.emplace_back(track.frame());
-          const int nGaps = size - 1;
-          streamTracks.insert(streamTracks.end(), nGaps, FrameTrack());
-        }
-        // transform deques to vectors & emulate truncation
-        auto limitTracks = next(streamTracks.begin(), min(setup_->numFrames(), (int)streamTracks.size()));
-        if (!enableTruncation_)
-          limitTracks = streamTracks.end();
-        streamAcceptedTracks[channelId] = StreamTrack(streamTracks.begin(), limitTracks);
-        streamLostTracks[channelId] = StreamTrack(limitTracks, streamTracks.end());
-        for (int layer = 0; layer < setup_->numLayers(); layer++) {
-          const int index = channelId * setup_->numLayers() + layer;
-          deque<FrameStub>& stubs = streamsStubs[layer];
-          auto limitStubs = next(stubs.begin(), min(setup_->numFrames(), (int)stubs.size()));
-          if (!enableTruncation_)
-            limitStubs = stubs.end();
-          streamAcceptedStubs[index] = StreamStub(stubs.begin(), limitStubs);
-          streamLostStubs[index] = StreamStub(limitStubs, stubs.end());
-        }
+      Handle<StreamsStub> handleStubs;
+      iEvent.getByToken<StreamsStub>(edGetTokenStubs_, handleStubs);
+      const StreamsStub& stubs = *handleStubs;
+      Handle<StreamsTrack> handleTracks;
+      iEvent.getByToken<StreamsTrack>(edGetTokenTracks_, handleTracks);
+      const StreamsTrack& tracks = *handleTracks;
+      for (int region = 0; region < setup_->numRegions(); region++) {
+        // object to fit tracks in a processing region
```

Comment:
Change comment to "Object containing data in a single processing region, used by subsequent track fit."

Reply:
at that point it does not contain data in a single processing region and itself is performing subsequent track fit. I find your comment more confusing then mine original one.

Reply:
I'm unconvinced. It is the KF that fits tracks, not the KFin.